tools/audio/aifc.c

```diff
@@ -0,0 +1,552 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (C) 2024 ZeldaRET
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "aifc.h"
+#include "util.h"
+
+#define CC4_CHECK(buf, str) \
+    ((buf)[0] == (str)[0] && (buf)[1] == (str)[1] && (buf)[2] == (str)[2] && (buf)[3] == (str)[3])
+
+#define CC4(c1, c2, c3, c4) (((c1) << 24) | ((c2) << 16) | ((c3) << 8) | (c4))
+
+#define FREAD(file, data, size)                                                                      \
+    do {                                                                                             \
+        if (fread((data), (size), 1, (file)) != 1) {                                                 \
+            error("[%s:%d] Could not read %lu bytes from file", __FILE__, __LINE__, (size_t)(size)); \
+        }                                                                                            \
+    } while (0)
+
+#define VADPCM_VER ((int16_t)1)
+
+#if 0
```

Comment:
would be nice-to-have if this was an arg to the program instead of needing to recompile. but not necessary.